app/helpers/session_helper.rb

```diff
@@ -0,0 +1,40 @@
+def user_at_email(email)
+	User.find_by_email(email)
+end
+
+def user_is_authenticated?(user, password)
+	return false if user.blank?
+	user.authenticate(password) ? true : false
+end
+
+def logged_in?
+	!!session[:current_user]
+end
+
+def session_set_current_user(user)
+	session[:current_user] = user
```

Comment:
You should not put complex ruby objects like an instance of `User` into your session. Just storing the `user_id` is sufficient.